sdk/azidentity/device_code_credential.go

```diff
@@ -158,7 +159,8 @@ func (c *DeviceCodeCredential) GetToken(ctx context.Context, opts policy.TokenRe
 		}
 		// if there is an error, check for an AADAuthenticationFailedError in order to check the status for token retrieval
 		// if the error is not an AADAuthenticationFailedError, then fail here since something unexpected occurred
-		if authRespErr := (*AADAuthenticationFailedError)(nil); errors.As(err, &authRespErr) && authRespErr.Message == "authorization_pending" {
+		var authFailed *AuthenticationFailedError
+		if errors.As(err, &authFailed) && strings.Contains(authFailed.msg, "authorization_pending") {
```

Comment:
This makes me a bit nervous, but given the change, I don't have a better solution.

Reply:
What if we kept the original error type but just didn't export it?  That way we can use these fields internally.

Reply:
The solution is: adopt MSAL 😉

Reply:
Oh yes of course.